### PR TITLE
feat(link-button): support isExternal prop

### PR DESCRIPTION
**Pull request by montezume**
#### Summary

Adds `isExternal` prop to `LinkButton`

#### Motivation

Some use cases in the MC (discount predicates for one) have this sort of external link, and this would make their code cleaner.

### CODE REVIEW

File: src/components/buttons/link-button/link-button.js [hunk 1]
```diff
@@ -1,65 +1,109 @@
...
+
+  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
+
+  ${props => !props.disabled && hoverStyles}
+`;
+
+const StyledReactRouterLink = createStyledComponent(ReactRouterLink);
+const StyledExternalLink = createStyledComponent('a');
+
+const componentProps = ['isExternal', 'to', 'label', 'isDisabled', 'iconLeft'];
```
Comment (tdeekens) [inline src/components/buttons/link-button/link-button.js]: This is odd to me. Is this an established pattern? I thought we could do something like `Object.keys(LinkButton.propTypes)` to determine the pass through props?

Comment (montezume) [reply src/components/buttons/link-button/link-button.js]: good call!

File: src/components/links/link/link.js [hunk 2]
```diff
@@ -19,10 +19,11 @@ const createStyledComponent = component => Styled(component)`
 const StyledReactRouterLink = createStyledComponent(ReactRouterLink);
 const StyledExternalLink = createStyledComponent('a');
 
-const componentProps = ['isExternal', 'to'];
-
 const Link = props => {
-  const remainingProps = getPassThroughProps(props, componentProps);
+  const remainingProps = getPassThroughProps(
+    props,
+    Object.keys(Link.propTypes)
```
Comment (montezume) [inline src/components/links/link/link.js]: thanks @tdeekens

File: src/components/buttons/link-button/link-button.js [hunk 3]
```diff
@@ -1,65 +1,110 @@
...
+          isDisabled={props.isDisabled}
+          label={props.label}
+        />
+      </StyledExternalLink>
+    );
+  }
+
+  return (
+    <StyledReactRouterLink
+      to={props.to}
```
Comment (jonnybel) [inline src/components/buttons/link-button/link-button.js]: This can be omitted from here because it's also passed in `remainingProps`, right?

Comment (montezume) [reply src/components/buttons/link-button/link-button.js]: no, it's removed from remainingProps because it's a propType right?

Comment (jonnybel) [reply src/components/buttons/link-button/link-button.js]: Ah yes, you're right!

**Comment by montezume** [comment]
Visual diff was just the extra proptype

**Pull request merged**